tests/unit/test_reservation.py

```diff
@@ -208,5 +215,402 @@ def test___heterogenous_session_infos___create_sessions___grouped_by_instrument_
+        grpc_session_infos[1].instrument_type_id = "nibar"
+        reservation = MultiSessionReservation(session_management_client, grpc_session_infos)
+        nifoo_info = stack.enter_context(reservation.create_session(_construct_session, "nifoo"))
+        nibar_info = stack.enter_context(reservation.create_session(_construct_session, "nibar"))
+
+        connection = reservation.get_connection(
+            fake_driver.Session, instrument_type_id=instrument_type_id
+        )
+
+        assert connection.pin_or_relay_name == pin_name
+        assert connection.site == site
+        assert connection.channel_name == channel_name
+        if instrument_type_id == "nifoo":
+            assert connection.session_info == nifoo_info
+        else:
+            assert connection.session_info == nibar_info
+
+
+def test___heterogenous_session_infos___get_connections___connections_returned(
+    session_management_client: Mock,
+) -> None:
+    with ExitStack() as stack:
+        grpc_session_infos = create_nifoo_session_infos(2)
+        grpc_session_infos[0].channel_mappings.add(pin_or_relay_name="Pin1", site=2, channel="3")
+        grpc_session_infos[1].channel_mappings.add(pin_or_relay_name="Pin4", site=5, channel="6")
+        grpc_session_infos[1].instrument_type_id = "nibar"
+        reservation = MultiSessionReservation(session_management_client, grpc_session_infos)
+        nifoo_info = stack.enter_context(reservation.create_session(_construct_session, "nifoo"))
+        nibar_info = stack.enter_context(reservation.create_session(_construct_session, "nibar"))
+
+        connections = reservation.get_connections(fake_driver.Session)
+
+        assert [conn.pin_or_relay_name for conn in connections] == ["Pin1", "Pin4"]
+        assert [conn.site for conn in connections] == [2, 5]
+        assert [conn.channel_name for conn in connections] == ["3", "6"]
+        assert [conn.session for conn in connections] == [nifoo_info.session, nibar_info.session]
+
+
+def test___wrong_session_type___get_connection___type_error_raised(
+    session_management_client: Mock,
+) -> None:
+    with ExitStack() as stack:
+        grpc_session_infos = create_nifake_session_infos(1)
+        grpc_session_infos[0].channel_mappings.add(pin_or_relay_name="Pin1", site=2, channel="3")
+        reservation = MultiSessionReservation(session_management_client, grpc_session_infos)
+        _ = stack.enter_context(reservation.create_session(_construct_session, "nifake"))
+
+        with pytest.raises(TypeError) as exc_info:
+            _ = reservation.get_connection(int)
+
+        assert "Incorrect type for session 'MySession0'." in exc_info.value.args[0]
+        assert (
+            "Expected <class 'int'>, got <class 'tests.utilities.fake_driver.Session'>."
+            in exc_info.value.args[0]
+        )
+
+
+def test___session_not_created___get_connection___type_error_raised(
```

Comment:
This got me thinking about LabVIEW. I thought something like this would work (it would just return an invalid session), but it doesn't. Casting an empty variant to an IVI session errors. So I'll have to file a bug for that.

Then the question becomes, should it be valid to get connections, even if you haven't created the session with our API? The output of get connections has more information than just the session. Should we allow you to get at that information, even if the session is invalid?

Reply:
Nevermind. It looks like you allow an empty session. So python does it right. I just have to fix LabVIEW to do something similar.